Multiple gRPC issues encountered when doing match query in parallel

## Description

In Grabl, `constructRepositorySummary` function constructs response for the owner page - statuses of 30 last commits for a given repo. The data for those commits is queried in parallel:

```
private static Map<String, Map<String, JsonArray>> constructRepositorySummary(GraknClient.Session session, List<Symbol.Commit> commits) {
try (GraknClient.Transaction tx = session.transaction(GraknClient.Transaction.Type.READ)) {
    List<List<Map<String, String>>> repoSummaryAnswer = commits.stream().parallel().map(commit -> {
        GraqlMatch.Filtered query = queryRepoSummary(commit);
        List<ConceptMap> conceptMaps = tx.query().match(query).collect(toList());
        return conceptMaps.stream().map(conceptMap -> {
            Map<String, String> map = new HashMap<>();
            map.put("pipeline-name", conceptMap.get("pipeline-name").asAttribute().asString().getValue());
            map.put("workflow-name", conceptMap.get("workflow-name").asAttribute().asString().getValue());
            map.put("workflow-status", conceptMap.get("workflow-status").asAttribute().asString().getValue());
            return map;
        }).collect(toList());
    }).collect(toList());

    Map<String, Map<String, JsonArray>> repoSummary = initRepositorySummaryMap(); // map(pipeline -> map(workflow -> array(status)))
    for (int i = 0; i < repoSummaryAnswer.size(); ++i) {
        List<Map<String, String>> commitSummary = repoSummaryAnswer.get(i);
        for (Map<String, String> workflow: commitSummary) {
            String pipelineName = workflow.get("pipeline-name");
            String workflowName = workflow.get("workflow-name");
            String workflowStatus = workflow.get("workflow-status");
            repoSummary.get(pipelineName).get(workflowName).set(Constants.COMMIT_COUNT_HEADS - i - 1, workflowStatus);
        }
    }

    List <String> statusesList = commits.stream().parallel().map(commit -> getDependencyAnalysisStatus(commit, tx)).collect(toList());
    for (int i = 0; i < statusesList.size(); ++i) {
        repoSummary.get("analysis").get("dependency-analysis").set(Constants.COMMIT_COUNT_HEADS - i - 1, statusesList.get(i));
    }
    return repoSummary;
}
```

Session is `graknClient.session(keyspace, GraknClient.Session.Type.DATA)`

Failing lines are: 

`List<List<Map<String, String>>> repoSummaryAnswer = commits.stream().parallel().map(commit  ...`
and the next `.parallel` call.


When trying to load the page, this function consistently (every time) throws some of the following erros: 

1) 
```
SEVERE: Exception while executing runnable io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1MessagesAvailable@4dd0d326
io.grpc.StatusRuntimeException: INTERNAL: Invalid protobuf byte sequence
        at io.grpc.Status.asRuntimeException(Status.java:524)
        at io.grpc.protobuf.lite.ProtoLiteUtils$MessageMarshaller.parse(ProtoLiteUtils.java:218)
        at io.grpc.protobuf.lite.ProtoLiteUtils$MessageMarshaller.parse(ProtoLiteUtils.java:118)
        at io.grpc.MethodDescriptor.parseRequest(MethodDescriptor.java:307)
        at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.messagesAvailableInternal(ServerCallImpl.java:309)
        at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.messagesAvailable(ServerCallImpl.java:292)
        at io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1MessagesAvailable.runInContext(ServerImpl.java:782)
        at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
        at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: com.google.protobuf.InvalidProtocolBufferException: Protocol message contained an invalid tag (zero).
        at com.google.protobuf.InvalidProtocolBufferException.invalidTag(InvalidProtocolBufferException.java:102)
        at com.google.protobuf.CodedInputStream$ArrayDecoder.readTag(CodedInputStream.java:628)
        at grakn.protocol.TransactionProto$Transaction$Req.<init>(TransactionProto.java:402)
        at grakn.protocol.TransactionProto$Transaction$Req.<init>(TransactionProto.java:369)
        at grakn.protocol.TransactionProto$Transaction$Req$1.parsePartialFrom(TransactionProto.java:3195)
        at grakn.protocol.TransactionProto$Transaction$Req$1.parsePartialFrom(TransactionProto.java:3189)
        at com.google.protobuf.AbstractParser.parseFrom(AbstractParser.java:86)
        at com.google.protobuf.AbstractParser.parseFrom(AbstractParser.java:48)
        at io.grpc.protobuf.lite.ProtoLiteUtils$MessageMarshaller.parseFrom(ProtoLiteUtils.java:223)
        at io.grpc.protobuf.lite.ProtoLiteUtils$MessageMarshaller.parse(ProtoLiteUtils.java:215)
        ... 10 more
```

2)

```
grakn.client.common.exception.GraknClientException: null
        at grakn.client.common.exception.GraknClientException.of(GraknClientException.java:48)
        at grakn.client.rpc.RPCTransaction$Response$Done$Error.read(RPCTransaction.java:323)
        at grakn.client.rpc.RPCTransaction$ResponseCollector.take(RPCTransaction.java:247)
        at grakn.client.rpc.ResponseIterator.computeNext(ResponseIterator.java:59)
        at com.google.common.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:145)
        at com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:140)
        at java.base/java.util.Iterator.forEachRemaining(Iterator.java:132)
        at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
        at grabl.concept.OwnerPage.lambda$constructRepositorySummary$3(OwnerPage.java:108)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
        at java.base/java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:952)
        at java.base/java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:926)
        at java.base/java.util.stream.AbstractTask.compute(AbstractTask.java:327)
        at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:746)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
        at java.base/java.util.concurrent.ForkJoinTask.doInvoke(ForkJoinTask.java:408)
        at java.base/java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:736)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateParallel(ReduceOps.java:919)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
        at grabl.concept.OwnerPage.constructRepositorySummary(OwnerPage.java:116)
        at grabl.concept.OwnerPage.constructRepository(OwnerPage.java:31)
        at grabl.service.Owner.constructOwnerPageChunk(Owner.java:60)
        at grabl.service.Owner.lambda$constructOwnerPage$0(Owner.java:50)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
        at java.base/java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:290)
        at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:746)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
        at java.base/java.util.concurrent.ForkJoinTask.doInvoke(ForkJoinTask.java:408)
        at java.base/java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:736)
        at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:159)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel(ForEachOps.java:173)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
        at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
        at grabl.service.Session.serveOwnerPage(Session.java:364)
        at grabl.service.Session.onMessage(Session.java:183)
        at akka.actor.typed.javadsl.BuiltReceive.receive(ReceiveBuilder.scala:184)
        at akka.actor.typed.javadsl.BuiltReceive.receiveMessage(ReceiveBuilder.scala:175)
        at akka.actor.typed.javadsl.Receive.receive(Receive.scala:53)
        at akka.actor.typed.javadsl.AbstractBehavior.receive(AbstractBehavior.scala:63)
        at akka.actor.typed.Behavior$.interpret(Behavior.scala:274)
        at akka.actor.typed.Behavior$.interpretMessage(Behavior.scala:230)
        at akka.actor.typed.internal.InterceptorImpl$$anon$2.apply(InterceptorImpl.scala:55)
        at akka.actor.typed.internal.SimpleSupervisor.aroundReceive(Supervision.scala:123)
        at akka.actor.typed.internal.InterceptorImpl.receive(InterceptorImpl.scala:83)
        at akka.actor.typed.Behavior$.interpret(Behavior.scala:274)
        at akka.actor.typed.Behavior$.interpretMessage(Behavior.scala:230)
        at akka.actor.typed.internal.adapter.ActorAdapter.handleMessage(ActorAdapter.scala:126)
        at akka.actor.typed.internal.adapter.ActorAdapter.aroundReceive(ActorAdapter.scala:106)
        at akka.actor.ActorCell.receiveMessage(ActorCell.scala:573)
        at akka.actor.ActorCell.invoke(ActorCell.scala:543)
        at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:269)
        at akka.dispatch.Mailbox.run(Mailbox.scala:230)
        at akka.dispatch.Mailbox.exec(Mailbox.scala:242)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```

3)
```
2021-02-17 10:41:13,724 [grakn-core-main::5] [ERROR] g.core.server.rpc.TransactionStream - [SRV14] Invalid Server Operation: The request message was not recognised.
grakn.core.common.exception.GraknException: [SRV14] Invalid Server Operation: The request message was not recognised.
        at grakn.core.common.exception.GraknException.of(GraknException.java:55)
        at grakn.core.server.rpc.TransactionRPC.handleRequest(TransactionRPC.java:124)
        at grakn.core.server.rpc.TransactionStream.handleRequest(TransactionStream.java:118)
        at grakn.core.server.rpc.TransactionStream.onNext(TransactionStream.java:82)
        at grakn.core.server.rpc.TransactionStream.onNext(TransactionStream.java:48)
        at io.grpc.stub.ServerCalls$StreamingServerCallHandler$StreamingServerCallListener.onMessage(ServerCalls.java:255)
        at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.messagesAvailableInternal(ServerCallImpl.java:309)
        at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.messagesAvailable(ServerCallImpl.java:292)
        at io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1MessagesAvailable.runInContext(ServerImpl.java:782)
        at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
        at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
```

4)

```
java.lang.IllegalArgumentException: java.lang.IllegalArgumentException: null frame before EOS
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
        at java.base/java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:600)
        at java.base/java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:678)
        at java.base/java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:737)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateParallel(ReduceOps.java:919)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
        at grabl.concept.OwnerPage.constructRepositorySummary(OwnerPage.java:116)
        at grabl.concept.OwnerPage.constructRepository(OwnerPage.java:31)
        at grabl.service.Owner.constructOwnerPageChunk(Owner.java:60)
        at grabl.service.Owner.lambda$constructOwnerPage$0(Owner.java:50)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
        at java.base/java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:290)
        at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:746)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
        at java.base/java.util.concurrent.ForkJoinTask.doInvoke(ForkJoinTask.java:408)
        at java.base/java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:736)
        at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:159)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel(ForEachOps.java:173)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
        at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
        at grabl.service.Session.serveOwnerPage(Session.java:364)
        at grabl.service.Session.onMessage(Session.java:183)
        at akka.actor.typed.javadsl.BuiltReceive.receive(ReceiveBuilder.scala:184)
        at akka.actor.typed.javadsl.BuiltReceive.receiveMessage(ReceiveBuilder.scala:175)
        at akka.actor.typed.javadsl.Receive.receive(Receive.scala:53)
        at akka.actor.typed.javadsl.AbstractBehavior.receive(AbstractBehavior.scala:63)
        at akka.actor.typed.Behavior$.interpret(Behavior.scala:274)
        at akka.actor.typed.Behavior$.interpretMessage(Behavior.scala:230)
        at akka.actor.typed.internal.InterceptorImpl$$anon$2.apply(InterceptorImpl.scala:55)
        at akka.actor.typed.internal.SimpleSupervisor.aroundReceive(Supervision.scala:123)
        at akka.actor.typed.internal.InterceptorImpl.receive(InterceptorImpl.scala:83)
        at akka.actor.typed.Behavior$.interpret(Behavior.scala:274)
        at akka.actor.typed.Behavior$.interpretMessage(Behavior.scala:230)
        at akka.actor.typed.internal.adapter.ActorAdapter.handleMessage(ActorAdapter.scala:126)
        at akka.actor.typed.internal.adapter.ActorAdapter.aroundReceive(ActorAdapter.scala:106)
        at akka.actor.ActorCell.receiveMessage(ActorCell.scala:573)
        at akka.actor.ActorCell.invoke(ActorCell.scala:543)
        at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:269)
        at akka.dispatch.Mailbox.run(Mailbox.scala:230)
        at akka.dispatch.Mailbox.exec(Mailbox.scala:242)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: java.lang.IllegalArgumentException: null frame before EOS
        at com.google.common.base.Preconditions.checkArgument(Preconditions.java:135)
        at io.grpc.internal.AbstractClientStream.deliverFrame(AbstractClientStream.java:186)
        at io.grpc.internal.MessageFramer.commitToSink(MessageFramer.java:352)
        at io.grpc.internal.MessageFramer.flush(MessageFramer.java:302)
        at io.grpc.internal.AbstractStream.flush(AbstractStream.java:75)
        at io.grpc.internal.ForwardingClientStream.flush(ForwardingClientStream.java:42)
        at io.grpc.internal.ClientCallImpl.sendMessageInternal(ClientCallImpl.java:593)
        at io.grpc.internal.ClientCallImpl.sendMessage(ClientCallImpl.java:564)
        at io.grpc.stub.ClientCalls$CallToStreamObserverAdapter.onNext(ClientCalls.java:364)
        at grakn.client.rpc.RPCTransaction.stream(RPCTransaction.java:175)
        at grakn.client.query.QueryManager.iterateQuery(QueryManager.java:156)
        at grakn.client.query.QueryManager.match(QueryManager.java:58)
        at grakn.client.query.QueryManager.match(QueryManager.java:52)
        at grabl.concept.OwnerPage.lambda$constructRepositorySummary$3(OwnerPage.java:108)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
        at java.base/java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:952)
        at java.base/java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:926)
        at java.base/java.util.stream.AbstractTask.compute(AbstractTask.java:327)
        at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:746)
        ... 5 common frames omitted
```

5)
```
io.netty.util.IllegalReferenceCountException: refCnt: 0, decrement: 1
        at io.netty.util.internal.ReferenceCountUpdater.toLiveRealRefCnt(ReferenceCountUpdater.java:74)
        at io.netty.util.internal.ReferenceCountUpdater.release(ReferenceCountUpdater.java:138)
        at io.netty.buffer.AbstractReferenceCountedByteBuf.release(AbstractReferenceCountedByteBuf.java:100)
        at io.netty.buffer.CompositeByteBuf$Component.free(CompositeByteBuf.java:1941)
        at io.netty.buffer.CompositeByteBuf.deallocate(CompositeByteBuf.java:2246)
        at io.netty.buffer.AbstractReferenceCountedByteBuf.handleRelease(AbstractReferenceCountedByteBuf.java:110)
        at io.netty.buffer.AbstractReferenceCountedByteBuf.release(AbstractReferenceCountedByteBuf.java:100)
        at io.netty.buffer.AbstractDerivedByteBuf.release0(AbstractDerivedByteBuf.java:94)
        at io.netty.buffer.AbstractDerivedByteBuf.rFeb 17, 2021 10:52:53 AM io.grpc.internal.SerializingExecutor run
SEVERE: Exception while executing runnable io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1MessagesAvailable@bd454cd
io.grpc.StatusRuntimeException: INTERNAL: Invalid protobuf byte sequence
        at io.grpc.Status.asRuntimeException(Status.java:524)
        at io.grpc.protobuf.lite.ProtoLiteUtils$MessageMarshaller.parse(ProtoLiteUtils.java:218)
        at io.grpc.protobuf.lite.ProtoLiteUtils$MessageMarshaller.parse(ProtoLiteUtils.java:118)
        at io.grpc.MethodDescriptor.parseRequest(MethodDescriptor.java:307)
        at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.messagesAvailableInternal(ServerCallImpl.java:309)
        at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.messagesAvailable(ServerCallImpl.java:292)
        at io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1MessagesAvailable.runInContext(ServerImpl.java:782)
        at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
        at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: com.google.protobuf.InvalidProtocolBufferException: Protocol message contained an invalid tag (zero).
        at com.google.protobuf.InvalidProtocolBufferException.invalidTag(InvalidProtocolBufferException.java:102)
        at com.google.protobuf.CodedInputStream$ArrayDecoder.readTag(CodedInputStream.java:628)
        at grakn.protocol.TransactionProto$Transaction$Req.<init>(TransactionProto.java:402)
        at grakn.protocol.TransactionProto$Transaction$Req.<init>(TransactionProto.java:369)
        at grakn.protocol.TransactionProto$Transaction$Req$1.parsePartialFrom(TransactionProto.java:3195)
        at grakn.protocol.TransactionProto$Transaction$Req$1.parsePartialFrom(TransactionProto.java:3189)
        at com.google.protobuf.AbstractParser.parseFrom(AbstractParser.java:86)
        at com.google.protobuf.AbstractParser.parseFrom(AbstractParser.java:48)
        at io.grpc.protobuf.lite.ProtoLiteUtils$MessageMarshaller.parseFrom(ProtoLiteUtils.java:223)
        at io.grpc.protobuf.lite.ProtoLiteUtils$MessageMarshaller.parse(ProtoLiteUtils.java:215)
        ... 10 more

elease(AbstractDerivedByteBuf.java:90)
        at io.netty.util.ReferenceCountUtil.release(ReferenceCountUtil.java:88)
        at io.netty.util.ReferenceCountUtil.safeRelease(ReferenceCountUtil.java:113)
        at io.netty.channel.ChannelOutboundBuffer.remove(ChannelOutboundBuffer.java:271)
        at io.netty.channel.ChannelOutboundBuffer.removeBytes(ChannelOutboundBuffer.java:352)
        at io.netty.channel.socket.nio.NioSocketChannel.doWrite(NioSocketChannel.java:431)
        at io.netty.channel.AbstractChannel$AbstractUnsafe.flush0(AbstractChannel.java:930)
        at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.flush0(AbstractNioChannel.java:354)
        at io.netty.channel.AbstractChannel$AbstractUnsafe.flush(AbstractChannel.java:897)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.flush(DefaultChannelPipeline.java:1372)
        at io.netty.channel.AbstractChannelHandlerContext.invokeFlush0(AbstractChannelHandlerContext.java:750)
        at io.netty.channel.AbstractChannelHandlerContext.invokeFlush(AbstractChannelHandlerContext.java:742)
        at io.netty.channel.AbstractChannelHandlerContext.flush(AbstractChannelHandlerContext.java:728)
        at io.netty.handler.codec.http2.Http2ConnectionHandler.flush(Http2ConnectionHandler.java:189)
        at io.netty.channel.AbstractChannelHandlerContext.invokeFlush0(AbstractChannelHandlerContext.java:750)
        at io.netty.channel.AbstractChannelHandlerContext.invokeFlush(AbstractChannelHandlerContext.java:742)
        at io.netty.channel.AbstractChannelHandlerContext.flush(AbstractChannelHandlerContext.java:728)
        at io.netty.channel.DefaultChannelPipeline.flush(DefaultChannelPipeline.java:967)
        at io.netty.channel.AbstractChannel.flush(AbstractChannel.java:242)
        at io.grpc.netty.WriteQueue.flush(WriteQueue.java:147)
        at io.grpc.netty.WriteQueue.access$000(WriteQueue.java:34)
        at io.grpc.netty.WriteQueue$1.run(WriteQueue.java:46)
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:834)

```

Also, sometimes UTF-8 related ones that I didn't copy. 

Queries themselves look like this

```
match
$commit isa commit, has symbol "VladGan/console@ec8270115acebb1d94aeaa72cbd57fa385e9a58b";
$_ (trigger: $commit, pipeline: $pipeline) isa pipeline-automation;
$pipeline has name $pipeline-name, has latest true;
$_ (pipeline: $pipeline, workflow: $workflow) isa pipeline-workflow;
$workflow has name $workflow-name, has status $workflow-status, has latest true;
get $pipeline-name, $workflow-name, $workflow-status;
```
(commit's symbol is a changeable part)

I verified that without `.parallel` same queries consistently work fine.

Link to the schema and data files:

https://www.dropbox.com/home/Engineering/Grabl/Issues/Grakn-2.0%20issues/275

--- a/rpc/ResponseIterator.java
+++ b/rpc/ResponseIterator.java
@@ -22,7 +22,6 @@ package grakn.client.rpc;
 import com.google.common.collect.AbstractIterator;
 import grakn.client.common.exception.GraknClientException;
 import grakn.protocol.TransactionProto;
-import io.grpc.stub.StreamObserver;
 
 import java.util.Iterator;
 import java.util.UUID;
@@ -35,12 +34,12 @@ import static grakn.common.util.Objects.className;
 class ResponseIterator<T> extends AbstractIterator<T> {
 
     private final UUID requestId;
-    private final StreamObserver<TransactionProto.Transaction.Req> requestObserver;
+    private final SynchronizedStreamObserver<TransactionProto.Transaction.Req> requestObserver;
     private final TransactionRPC.ResponseCollector.Multiple responseCollector;
     private final Function<TransactionProto.Transaction.Res, Stream<T>> transformResponse;
     private Iterator<T> currentIterator;
 
-    ResponseIterator(UUID requestId, StreamObserver<TransactionProto.Transaction.Req> requestObserver,
+    ResponseIterator(UUID requestId, SynchronizedStreamObserver<TransactionProto.Transaction.Req> requestObserver,
                      TransactionRPC.ResponseCollector.Multiple responseCollector, Function<TransactionProto.Transaction.Res, Stream<T>> transformResponse) {
         this.requestId = requestId;
         this.transformResponse = transformResponse;

--- a/rpc/SynchronizedStreamObserver.java
+++ b/rpc/SynchronizedStreamObserver.java
@@ -1,0 +1,19 @@
+package grakn.client.rpc;
+
+import io.grpc.stub.StreamObserver;
+
+public class SynchronizedStreamObserver<T> {
+    private final StreamObserver<T> s;
+
+    SynchronizedStreamObserver(StreamObserver<T> stream) {
+        s = stream;
+    }
+
+    synchronized void onNext(T value) {
+        s.onNext(value);
+    }
+
+    synchronized void onCompleted() {
+        s.onCompleted();
+    }
+}

--- a/rpc/SynchronizedStreamObserver.java
+++ b/rpc/SynchronizedStreamObserver.java
@@ -1,3 +1,22 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
 package grakn.client.rpc;
 
 import io.grpc.stub.StreamObserver;

--- a/rpc/TransactionRPC.java
+++ b/rpc/TransactionRPC.java
@@ -57,7 +57,7 @@ public class TransactionRPC implements Transaction {
     private final ConceptManager conceptManager;
     private final LogicManager logicManager;
     private final QueryManager queryManager;
-    private final StreamObserver<TransactionProto.Transaction.Req> requestObserver;
+    private final SynchronizedStreamObserver<TransactionProto.Transaction.Req> requestObserver;
     private final ResponseCollectors collectors;
     private final int networkLatencyMillis;
     private final AtomicBoolean isOpen;
@@ -72,7 +72,7 @@ public class TransactionRPC implements Transaction {
 
             // Opening the StreamObserver exposes these atomics to another thread, so we must initialize them first.
             isOpen = new AtomicBoolean(true);
-            requestObserver = GraknGrpc.newStub(session.channel()).transaction(responseObserver());
+            requestObserver = new SynchronizedStreamObserver<>(GraknGrpc.newStub(session.channel()).transaction(responseObserver()));
             final TransactionProto.Transaction.Req.Builder openRequest = TransactionProto.Transaction.Req.newBuilder()
                     .putAllMetadata(tracingData())
                     .setOpenReq(TransactionProto.Transaction.Open.Req.newBuilder()
@@ -174,6 +174,7 @@ public class TransactionRPC implements Transaction {
         collectors.put(requestId, responseCollector);
         requestObserver.onNext(request.build());
         final ResponseIterator<T> responseIterator = new ResponseIterator<>(requestId, requestObserver, responseCollector, transformResponse);
+        System.out.println(Thread.currentThread().getId() + "  ResponseIterator      requestId = " + requestId + "  request = " + request );
         return StreamSupport.stream(((Iterable<T>) () -> responseIterator).spliterator(), false);
     }
 

--- a/rpc/TransactionRPC.java
+++ b/rpc/TransactionRPC.java
@@ -174,7 +174,6 @@ public class TransactionRPC implements Transaction {
         collectors.put(requestId, responseCollector);
         requestObserver.onNext(request.build());
         final ResponseIterator<T> responseIterator = new ResponseIterator<>(requestId, requestObserver, responseCollector, transformResponse);
-        System.out.println(Thread.currentThread().getId() + "  ResponseIterator      requestId = " + requestId + "  request = " + request );
         return StreamSupport.stream(((Iterable<T>) () -> responseIterator).spliterator(), false);
     }
 

--- a/test/integration/ClientQueryTest.java
+++ b/test/integration/ClientQueryTest.java
@@ -41,7 +41,6 @@ import java.util.concurrent.TimeoutException;
 import java.util.function.Consumer;
 import java.util.stream.Stream;
 
-import static grakn.client.GraknClient.Transaction.Type.READ;
 import static grakn.client.GraknClient.Transaction.Type.WRITE;
 import static graql.lang.Graql.and;
 import static graql.lang.Graql.rel;
@@ -86,10 +85,17 @@ public class ClientQueryTest {
                     type("name").sub("attribute").value(GraqlArg.ValueType.STRING),
                     type("lion").sub("entity").owns("name").plays("mating", "male-partner").plays("mating", "female-partner").plays("child-bearing", "offspring").plays("parentship", "parent").plays("parentship", "child")
             );
-
+            final GraqlDefine ruleQuery = Graql.define(rule("infer-parentship-from-mating-and-child-bearing")
+                    .when(and(
+                            rel("male-partner", var("male")).rel("female-partner", var("female")).isa("mating"),
+                            var("childbearing").rel("child-bearer").rel("offspring", var("offspring")).isa("child-bearing")))
+                    .then(rel("parent", var("male"))
+                            .rel("parent", var("female"))
+                            .rel("child", var("offspring")).isa("parentship")));
             LOG.info("clientJavaE2E() - define a schema...");
             LOG.info("clientJavaE2E() - '" + defineQuery + "'");
             tx.query().define(defineQuery);
+            tx.query().define(ruleQuery);
             tx.commit();
             LOG.info("clientJavaE2E() - done.");
         }, Session.Type.SCHEMA);

--- a/test/integration/ClientQueryTest.java
+++ b/test/integration/ClientQueryTest.java
@@ -22,11 +22,13 @@ package grakn.client.test.integration;
 import grakn.client.GraknClient;
 import grakn.client.GraknClient.Session;
 import grakn.client.GraknClient.Transaction;
+import grakn.client.concept.answer.ConceptMap;
 import grakn.common.test.server.GraknCoreRunner;
 import graql.lang.Graql;
 import graql.lang.common.GraqlArg;
 import graql.lang.query.GraqlDefine;
 import graql.lang.query.GraqlInsert;
+import graql.lang.query.GraqlMatch;
 import org.junit.AfterClass;
 import org.junit.BeforeClass;
 import org.junit.Test;
@@ -34,15 +36,19 @@ import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
+import java.util.List;
 import java.util.concurrent.TimeoutException;
 import java.util.function.Consumer;
+import java.util.stream.Stream;
 
+import static grakn.client.GraknClient.Transaction.Type.READ;
 import static grakn.client.GraknClient.Transaction.Type.WRITE;
 import static graql.lang.Graql.and;
 import static graql.lang.Graql.rel;
 import static graql.lang.Graql.rule;
 import static graql.lang.Graql.type;
 import static graql.lang.Graql.var;
+import static java.util.stream.Collectors.toList;
 
 @SuppressWarnings("Duplicates")
 public class ClientQueryTest {
@@ -66,6 +72,7 @@ public class ClientQueryTest {
         grakn.stop();
     }
 
+
     @Test
     public void applicationTest() {
         LOG.info("clientJavaE2E() - starting client-java E2E...");
@@ -80,17 +87,9 @@ public class ClientQueryTest {
                     type("lion").sub("entity").owns("name").plays("mating", "male-partner").plays("mating", "female-partner").plays("child-bearing", "offspring").plays("parentship", "parent").plays("parentship", "child")
             );
 
-            final GraqlDefine ruleQuery = Graql.define(rule("infer-parentship-from-mating-and-child-bearing")
-                     .when(and(
-                             rel("male-partner", var("male")).rel("female-partner", var("female")).isa("mating"),
-                             var("childbearing").rel("child-bearer").rel("offspring", var("offspring")).isa("child-bearing")))
-                     .then(rel("parent", var("male"))
-                             .rel("parent", var("female"))
-                             .rel("child", var("offspring")).isa("parentship")));
             LOG.info("clientJavaE2E() - define a schema...");
             LOG.info("clientJavaE2E() - '" + defineQuery + "'");
             tx.query().define(defineQuery);
-            tx.query().define(ruleQuery);
             tx.commit();
             LOG.info("clientJavaE2E() - done.");
         }, Session.Type.SCHEMA);
@@ -220,6 +219,237 @@ public class ClientQueryTest {
 //        });
 
         LOG.info("clientJavaE2E() - client-java E2E test done.");
+    }
+
+    private String[] commitSHAs() {
+        return new String[] {
+                // queried commits
+                "VladGan/console@4bdc38acb87f9fd2fbdb7cbcf2bcc93837382cab",
+                "VladGan/console@b5ecd4707ce425d7d2d4d0b0d53420cb46e8ce52",
+                "VladGan/console@b16788637949c6b4c2a3a4bacc8da101bf838b38",
+                "VladGan/console@8e996fdf8d802d270385ac3bc7cbf5fa77ac0583",
+                "VladGan/console@1ff6651afa7abf43b5bdd3b1903e489d279e3dc6",
+                "VladGan/console@6d3ceda79eb3e3dc86d266095b613a53fb083d30",
+                "VladGan/console@23da5b400e32805c29f41671ff3f92ef48eafcf8",
+                // not queried commits
+                "VladGan/console@0000000000000000000000000000000000000000",
+                "VladGan/console@1111111111111111111111111111111111111111",
+                "VladGan/console@2222222222222222222222222222222222222222",
+        };
+    }
+
+    @Test
+    public void parallelQueriesInTransactionTest() {
+        localhostGraknTx(tx -> {
+            final GraqlDefine defineQuery = Graql.define(
+                type("symbol").sub("attribute").value(GraqlArg.ValueType.STRING),
+                type("name").sub("attribute").value(GraqlArg.ValueType.STRING),
+                type("status").sub("attribute").value(GraqlArg.ValueType.STRING),
+                type("latest").sub("attribute").value(GraqlArg.ValueType.BOOLEAN),
+
+                type("commit").sub("entity")
+                        .owns("symbol")
+                        .plays("pipeline-automation", "trigger"),
+                type("pipeline").sub("entity")
+                        .owns("name")
+                        .owns("latest")
+                        .plays("pipeline-workflow", "pipeline")
+                        .plays("pipeline-automation", "pipeline"),
+                type("workflow").sub("entity")
+                        .owns("name")
+                        .owns("status")
+                        .owns("latest")
+                        .plays("pipeline-workflow", "workflow"),
+
+                type("pipeline-workflow").sub("relation")
+                        .relates("pipeline").relates("workflow"),
+                type("pipeline-automation").sub("relation")
+                        .relates("pipeline").relates("trigger")
+            );
+
+            LOG.info("clientJavaE2E() - define a schema...");
+            LOG.info("clientJavaE2E() - '" + defineQuery + "'");
+            tx.query().define(defineQuery);
+            tx.commit();
+            LOG.info("clientJavaE2E() - done.");
+        }, Session.Type.SCHEMA);
+
+
+
+        localhostGraknTx(tx -> {
+            final String[] commits = commitSHAs();
+            final GraqlInsert insertCommitQuery = Graql.insert(
+                    var().isa("commit").has("symbol", commits[0]),
+                    var().isa("commit").has("symbol", commits[1]),
+                    var().isa("commit").has("symbol", commits[3]),
+                    var().isa("commit").has("symbol", commits[4]),
+                    var().isa("commit").has("symbol", commits[5]),
+                    var().isa("commit").has("symbol", commits[6]),
+                    var().isa("commit").has("symbol", commits[7]),
+                    var().isa("commit").has("symbol", commits[8]),
+                    var().isa("commit").has("symbol", commits[9])
+            );
+
+            LOG.info("clientJavaE2E() - insert commit data...");
+            LOG.info("clientJavaE2E() - '" + insertCommitQuery + "'");
+            tx.query().insert(insertCommitQuery);
+            tx.commit();
+            LOG.info("clientJavaE2E() - done.");
+        });
+
+        localhostGraknTx(tx -> {
+            final GraqlInsert insertWorkflowQuery = Graql.insert(
+                    var().isa("workflow")
+                            .has("name", "workflow-A")
+                            .has("status", "running")
+                            .has("latest", true),
+                    var().isa("workflow")
+                            .has("name", "workflow-B")
+                            .has("status", "finished")
+                            .has("latest", false)
+            );
+
+            LOG.info("clientJavaE2E() - insert workflow data...");
+            LOG.info("clientJavaE2E() - '" + insertWorkflowQuery + "'");
+            tx.query().insert(insertWorkflowQuery);
+            tx.commit();
+            LOG.info("clientJavaE2E() - done.");
+        });
+
+        localhostGraknTx(tx -> {
+            final GraqlInsert insertPipelineQuery = Graql.insert(
+                    var().isa("pipeline")
+                            .has("name", "pipeline-A")
+                            .has("latest", true),
+                    var().isa("pipeline")
+                            .has("name", "pipeline-B")
+                            .has("latest", false)
+            );
+
+            LOG.info("clientJavaE2E() - insert pipeline data...");
+            LOG.info("clientJavaE2E() - '" + insertPipelineQuery + "'");
+            tx.query().insert(insertPipelineQuery);
+            tx.commit();
+            LOG.info("clientJavaE2E() - done.");
+        });
+
+        localhostGraknTx(tx -> {
+            final String[] commitShas = commitSHAs();
+            LOG.info("clientJavaE2E() - inserting pipeline-automation relations...");
+
+            for (int i = 0; i < commitShas.length / 2; i++) {
+                final GraqlInsert insertPipelineAutomationQuery = Graql.match(
+                        var("commit").isa("commit").has("symbol", commitShas[i]),
+                        var("pipeline").isa("pipeline").has("name", "pipeline-A")
+                )
+                        .insert(
+                                rel("pipeline", "pipeline").rel("trigger", "commit").isa("pipeline-automation")
+                        );
+                LOG.info("clientJavaE2E() - '" + insertPipelineAutomationQuery + "'");
+                final List<ConceptMap> x = tx.query().insert(insertPipelineAutomationQuery).collect(toList());
+            }
+
+
+            for (int i = commitShas.length / 2; i < commitShas.length; i++) {
+                final GraqlInsert insertPipelineAutomationQuery = Graql.match(
+                        var("commit").isa("commit").has("symbol", commitShas[i]),
+                        var("pipeline").isa("pipeline").has("name", "pipeline-B")
+                )
+                        .insert(
+                                rel("pipeline", "pipeline").rel("trigger", "commit").isa("pipeline-automation")
+                        );
+                LOG.info("clientJavaE2E() - '" + insertPipelineAutomationQuery + "'");
+                final List<ConceptMap> x = tx.query().insert(insertPipelineAutomationQuery).collect(toList());
+            }
+
+            tx.commit();
+
+            LOG.info("clientJavaE2E() - done.");
+        });
+
+        localhostGraknTx(tx -> {
+            LOG.info("clientJavaE2E() - inserting pipeline-automation relations...");
+
+            final GraqlInsert insertPipelineWorkflowQuery = Graql.match(
+                    var("pipelineA").isa("pipeline").has("name", "pipeline-A"),
+                    var("workflowA").isa("workflow").has("name", "workflow-A"),
+                    var("pipelineB").isa("pipeline").has("name", "pipeline-B"),
+                    var("workflowB").isa("workflow").has("name", "workflow-B")
+            )
+                    .insert(
+                            rel("pipeline", "pipelineA").rel("workflow", "workflowA").isa("pipeline-workflow"),
+                            rel("pipeline", "pipelineB").rel("workflow", "workflowB").isa("pipeline-workflow")
+                    );
+            LOG.info("clientJavaE2E() - '" + insertPipelineWorkflowQuery + "'");
+            final List<ConceptMap> x = tx.query().insert(insertPipelineWorkflowQuery).collect(toList());
+
+            tx.commit();
+
+            LOG.info("clientJavaE2E() - done.");
+        });
+
+        String[] queries = {
+                "match\n" +
+                        "$commit isa commit, has symbol \"VladGan/console@4bdc38acb87f9fd2fbdb7cbcf2bcc93837382cab\";\n" +
+                        "$_ (trigger: $commit, pipeline: $pipeline) isa pipeline-automation;\n" +
+                        "$pipeline has name $pipeline-name, has latest true;\n" +
+                        "$_ (pipeline: $pipeline, workflow: $workflow) isa pipeline-workflow;\n" +
+                        "$workflow has name $workflow-name, has status $workflow-status, has latest true;\n" +
+                        "get $pipeline-name, $workflow-name, $workflow-status;",
+                "match\n" +
+                        "$commit isa commit, has symbol \"VladGan/console@b5ecd4707ce425d7d2d4d0b0d53420cb46e8ce52\";\n" +
+                        "$_ (trigger: $commit, pipeline: $pipeline) isa pipeline-automation;\n" +
+                        "$pipeline has name $pipeline-name, has latest true;\n" +
+                        "$_ (pipeline: $pipeline, workflow: $workflow) isa pipeline-workflow;\n" +
+                        "$workflow has name $workflow-name, has status $workflow-status, has latest true;\n" +
+                        "get $pipeline-name, $workflow-name, $workflow-status;",
+                "match\n" +
+                        "$commit isa commit, has symbol \"VladGan/console@b16788637949c6b4c2a3a4bacc8da101bf838b38\";\n" +
+                        "$_ (trigger: $commit, pipeline: $pipeline) isa pipeline-automation;\n" +
+                        "$pipeline has name $pipeline-name, has latest true;\n" +
+                        "$_ (pipeline: $pipeline, workflow: $workflow) isa pipeline-workflow;\n" +
+                        "$workflow has name $workflow-name, has status $workflow-status, has latest true;\n" +
+                        "get $pipeline-name, $workflow-name, $workflow-status;",
+                "match\n" +
+                        "$commit isa commit, has symbol \"VladGan/console@8e996fdf8d802d270385ac3bc7cbf5fa77ac0583\";\n" +
+                        "$_ (trigger: $commit, pipeline: $pipeline) isa pipeline-automation;\n" +
+                        "$pipeline has name $pipeline-name, has latest true;\n" +
+                        "$_ (pipeline: $pipeline, workflow: $workflow) isa pipeline-workflow;\n" +
+                        "$workflow has name $workflow-name, has status $workflow-status, has latest true;\n" +
+                        "get $pipeline-name, $workflow-name, $workflow-status;",
+                "match\n" +
+                        "$commit isa commit, has symbol \"VladGan/console@1ff6651afa7abf43b5bdd3b1903e489d279e3dc6\";\n" +
+                        "$_ (trigger: $commit, pipeline: $pipeline) isa pipeline-automation;\n" +
+                        "$pipeline has name $pipeline-name, has latest true;\n" +
+                        "$_ (pipeline: $pipeline, workflow: $workflow) isa pipeline-workflow;\n" +
+                        "$workflow has name $workflow-name, has status $workflow-status, has latest true;\n" +
+                        "get $pipeline-name, $workflow-name, $workflow-status;",
+                "match\n" +
+                        "$commit isa commit, has symbol \"VladGan/console@6d3ceda79eb3e3dc86d266095b613a53fb083d30\";\n" +
+                        "$_ (trigger: $commit, pipeline: $pipeline) isa pipeline-automation;\n" +
+                        "$pipeline has name $pipeline-name, has latest true;\n" +
+                        "$_ (pipeline: $pipeline, workflow: $workflow) isa pipeline-workflow;\n" +
+                        "$workflow has name $workflow-name, has status $workflow-status, has latest true;\n" +
+                        "get $pipeline-name, $workflow-name, $workflow-status;",
+                "match\n" +
+                        "$commit isa commit, has symbol \"VladGan/console@23da5b400e32805c29f41671ff3f92ef48eafcf8\";\n" +
+                        "$_ (trigger: $commit, pipeline: $pipeline) isa pipeline-automation;\n" +
+                        "$pipeline has name $pipeline-name, has latest true;\n" +
+                        "$_ (pipeline: $pipeline, workflow: $workflow) isa pipeline-workflow;\n" +
+                        "$workflow has name $workflow-name, has status $workflow-status, has latest true;\n" +
+                        "get $pipeline-name, $workflow-name, $workflow-status;"
+        };
+
+        localhostGraknTx(tx -> {
+            LOG.info("clientJavaE2E() - inserting pipeline-automation relations...");
+
+            Stream.of(queries).parallel().forEach(x -> {
+                GraqlMatch q = Graql.parseQuery(x).asMatch();
+                final List<ConceptMap> res = tx.query().match(q).collect(toList());
+            });
+
+            LOG.info("clientJavaE2E() - done.");
+        });
     }
 
     private String[] lionNames() {

